sycl/source/detail/plugin.hpp

```diff
@@ -31,6 +31,10 @@ class plugin {
 
   const RT::PiPlugin &getPiPlugin() const { return MPlugin; }
 
+  bool isBackendType(Backend backend) const {
+    return MPlugin.backend == backend;
+  }
```

Comment:
Seems awkward to have a comparison method inside a plugin, why not simply return the Plugin type and let the caller do the comparison?
Simply have a 
```cpp
Backend getBackendType() const noexcept {
   return MPlugin.backend;
}
```

Reply:
+1 for returning the Backend.

Also we need this info be available externally. Could you add a method to return the BE via, say platform.get_info<info::platform::backend>?

Reply:
yes sure. Actually, as I'm at it already, maybe there is a more agnostic way than to have the plugin types hard-coded in an enum in "pi.h"?

what if - for example - there was a `char* backend` which we can return as `std::string getBackendType()` and `std::string platform.get_info<>`? Then the plugin interface stays a bit more general and we would compare `ContextImpl->getPlugin()->getBackendType() == "OpenCL"`

On the other hand i found this: [Sycl-Shared](https://github.com/KhronosGroup/SYCL-Shared/blob/master/proposals/sycl_generalization.md) and they describe to even have an enum at `sycl::backend` to compare with...

Reply:
>  have an enum at sycl::backend
That's right. It is coming very soon in this PR:
https://github.com/intel/llvm/pull/1332/files#diff-f42957493751abda949e943bbb84b93d
Let's use it in these interfaces.

Reply:
ok, thanks. Then I'll wait for that PR to be merged and base off it.

regarding the platform.get_info<> it should return an std::string in the end...?

Reply:
I think it should return the "backend" not a string. But note that the BE generalization proposal defines get_backend() method instead (maybe in addition?): https://github.com/KhronosGroup/SYCL-Shared/blob/master/proposals/sycl_generalization.md

Reply:
+1 on returning the backend, not a string.

We don't have a `get_info` query for backends on the generalization proposal, but can be easily added if that is useful.

Reply:
yep, sure. thanks. i will stick to whats in the generalization for now.

```
// Returns the backend to which this platform belongs
backend get_backend() const noexcept;
```

i will wait for #1332 to be merged and start from there.